### PR TITLE
Add storage to resourceread

### PR DESCRIPTION
**Pull request by wongma7**
for reading StorageClasses

### CODE REVIEW

**Comment by wongma7** [comment]
Almost every interesting field on storageclass is immutable so i didn't think `apply` very useful. however for consistency it will be good to have, will add it to this PR tomorrow

**Comment by wongma7** [comment]
@deads2k added an `Apply`, PTAL

File: pkg/operator/resource/resourceapply/storage.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+	contentSame := equality.Semantic.DeepEqual(existing, required)
+	if contentSame && !*modified {
+		return existing, false, nil
+	}
+
+	objectMeta := existing.ObjectMeta.DeepCopy()
+	existing = required.DeepCopy()
+	existing.ObjectMeta = *objectMeta
+
+	// if provisioner, parameters, reclaimpolicy, or volumebindingmode are different the update will fail
```
Comment (deads2k) [inline pkg/operator/resource/resourceapply/storage.go]: make a TODO that would cause this to delete and recreate the storage class.

Comment (wongma7) [reply pkg/operator/resource/resourceapply/storage.go]: comment amended to TODO

**Pull request merged**